src/TemplateBinding.js

```diff
@@ -1058,6 +1106,20 @@
     updateValue: function(value) {
       if (!this.deps.repeat)
         value = [value];
+      else if (Object.prototype.toString.call(value)  === '[object Object]') {
+        if (this.objectArrayProxy !== undefined) {
+          if (value !== this.objectArrayProxy.values) {
+            this.objectArrayProxy.closeObserver();
```

Comment:
it seems like some duplication between closeObserver and resetKeys_ (which itself knows how to do closeObserver), is there a way to factor that better? ideally the code here is more like:

```js
if (!this.keysObserver) {
  this.keysObserver = new ...
}
// start or reset observation on this.keysObserver
```